src/client/Microsoft.Identity.Client/Internal/Constants.cs

```diff
@@ -42,6 +42,7 @@ internal static class Constants
         public const string SshCertAuthHeaderPrefix = "SshCert";
 
         public const string ManagedIdentityClientId = "client_id";
```

Comment:
I think you should change this to "ManagedIdentityClientId2019".

Reply:
I think the key here is to NOT think "api 2017 always uses clientid", "api 2019 always client_id", "managed identity Foo uses api 2019 and managed identity Bar also uses api 2019 so they are likely the same". No. Each managed identity provider *models* their API from a similar template but their implementations are effectively different, possibly not sharing the same code base.

That's what I meant the other day by saying "ideally MSAL shall implement each managed identity protocol in their own code path so that they don't share the same implementation". @Robbie-Microsoft